clients/app/src/core/stores/generateCRUD/index.ts

```diff
@@ -3,130 +3,212 @@ import type { SetState, GetState } from 'zustand/vanilla';
+>(
+  params: ParamsType<TData, TKey>
+) => {
+  // @ts-ignore
+  const includeKeys = fillStateKeys(params.keys || []);
+
   // #region state
-  const buildInitialState = (): BaseStateDataInterface<T> => ({
-    list: {
-      status: 'loading',
-      error: null,
-      data: null,
-      pagination: {
-        total: null,
-        limit: 0,
-        offset: 0,
-      },
-    },
-
-    single: {
-      status: 'loading',
-      error: null,
-      data: null,
-    },
-  });
+  const buildInitialState = (): TExtractState<TData, TKey> => {
+    // @ts-ignore
+    const state: TExtractState<TData, TKey> = {};
+
+    if (includeKeys.list) {
+      (state as TExtractState<TData, 'list'>).list = {
+        status: 'loading',
+        error: null,
+        data: null,
+        pagination: {
+          total: null,
+          limit: 0,
+          offset: 0,
+        },
+      };
+    }
+
+    if (includeKeys.single) {
+      (state as TExtractState<TData, 'single'>).single = {
+        status: 'loading',
+        error: null,
+        data: null,
+      };
+    }
+
+    return state;
+  };
   // #endregion
 
   // #region actions
   const buildActions = (
-    setState: SetState<BaseStoreInterface<T>>,
-    getState: GetState<BaseStoreInterface<T>>
-  ): BaseStateActionInterface<T> => ({
-    // #region action - getAll
-    async getAll(): Promise<ListStateDataInterface<T>> {
-      const list = getState().list;
-
-      const itemsLimit = params.limit || defaultLimit;
-      const itemsCount = list.data?.length || 0;
-
-      if (list.pagination.total === itemsCount) {
-        return list;
-      }
-
-      list.status = 'loading';
-      setState({ list });
+    setStateCall: SetState<TExtractStateAndActions<TData, TKey>>,
+    getStateCall: GetState<TExtractStateAndActions<TData, TKey>>
+  ): TExtractAction<TData, TKey> => {
+    // @ts-ignore
+    const actions: TExtractAction<TData, TKey> = {};
+
+    if (includeKeys.list) {
+      const getState = getStateCall as unknown as StateGetterCallType<TData, 'list'>;
+      const setState = setStateCall as unknown as StateSetterCallType<TData, 'list'>;
+      const callbacks = params as unknown as TExtractCallback<TData, 'list'>;
+
+      (actions as TExtractAction<TData, 'list'>).getAll = async (): Promise<
+        ListStateDataInterface<TData>
+      > => {
+        const list = getState().list;
+
+        const itemsLimit = params.limit || defaultLimit;
+        const itemsCount = list.data?.length || 0;
+
+        if (list.pagination.total === itemsCount) {
+          return list;
+        }
 
-      try {
-        const result = await params.loadAll(itemsLimit, itemsCount);
-        if (!result) {
-          throw new Error('SDK returned empty result.');
+        list.status = 'loading';
+        setState({ list });
+
+        try {
+          const result = await callbacks.loadAll(itemsLimit, itemsCount);
+          if (!result) {
+            throw new Error('SDK returned empty result.');
```

Comment:
Are you sure  `SDK returned empty result` is the best message here?,  maybe `[collbackName] returned empty result`?.
I think the users of this module, will not know why `SDK`? which `SDK`, and absolutely have no idea, because they provided just a function. And this module doesn't know  about SDK too.  
Thoughts?

Reply:
agree with you.

Reply:
I think this message is described more accurately, e.g.:
`Interceptor 'loadAll(limit: 50, offset: 200)' returned empty result.`

Reply:
The name is enough I think, later will define the result interface need to be returned from the interceptor, and the message will be the Interceptor should return Interceptor or something like this